Add "Upgrading from Godot 4.3 to Godot 4.4" page (#10546)
godotengine/godot-docs: tutorials/migrating/upgrading_to_godot_4.4.rst

Method ``popup_node_selector`` adds a new ``current_value`` optional parameter                                            |✔️|                 |✔️ with compat|      |✔️ with compat|      `GH-94323`_
Method ``popup_property_selector`` adds a new ``current_value`` optional parameter                                        |✔️|                 |✔️ with compat|      |✔️ with compat|      `GH-94323`_
**EditorSceneFormatImporter**
Method ``_get_import_flags`` removed                                                                                      |❌|                 |❌|                  |❌|                  `GH-101531`_
========================================================================================================================  ===================  ====================  ====================  ============

.. note::

    The method ``_get_import_flags`` was never used by the engine, since it was open source. So it was removed
    despite the compat breakage as there's no way for users to rely on this affecting engine behavior.

Behavior changes
----------------

Rendering
~~~~~~~~~

.. note::

>     The ``VisualShaderNodeVec4Constant`` shader node had its type changed to ``Vector4``. Users need to recreate
>     the values in their constants.

Reviewer: ```suggestion
    The ``VisualShaderNodeVec4Constant`` shader node had its input type changed to ``Vector4``. Users need to
    recreate the values in their constants.
```
If keeping I'd clarify we mean "input type" here, otherwise reads as confusing

Method ``popup_node_selector`` adds a new ``current_value`` optional parameter                                            |✔️|                 |✔️ with compat|      |✔️ with compat|      `GH-94323`_
Method ``popup_property_selector`` adds a new ``current_value`` optional parameter                                        |✔️|                 |✔️ with compat|      |✔️ with compat|      `GH-94323`_
**EditorSceneFormatImporter**
Method ``_get_import_flags`` removed                                                                                      |❌|                 |❌|                  |❌|                  `GH-101531`_
**EditorTranslationParserPlugin**
Method ``_parse_file`` changes return type to ``Array`` and removes ``msgids`` and ``msgids_context_plural`` parameters   |❌|                 |❌|                  |❌|                  `GH-99297`_
========================================================================================================================  ===================  ====================  ====================  ============

.. note::

>     The method ``_get_import_flags`` was never used by the engine, since it was open source. So it was removed
>     despite the compat breakage as there's no way for users to rely on this affecting engine behavior.

Reviewer: ```suggestion
    The method ``_get_import_flags`` was never used by the engine, since the engine became open source.
    It was removed despite the compatibility breakage as there's no way for users to rely on this affecting
    engine behavior.
```

Behavior changes
----------------

Core
~~~~

.. note::

    The ``Curve`` resource now enforces its value range, so ``min_value`` and ``max_value`` need to be changed
    if any of the points fall outside of the default ``[0, 1]`` range.

Rendering
~~~~~~~~~

.. note::

    The ``VisualShaderNodeVec4Constant`` shader node had its input type changed to ``Vector4``. Users need to
    recreate the values in their constants.

CSG
~~~

.. note::

    The CSG implementation now uses Emmett Lalish's `Manifold <https://github.com/elalish/manifold>`_ library (`GH-94321`_).
    The new implementation is more consistent with manifold definitions and fixes a number of bugs and stability

>     issues. As a result, non-manifold meshes are no longer supported. Use ``MeshInstance3D`` for rendering quads
>     and other non-manifold geometry.

Reviewer: ```suggestion
    issues. As a result, non-manifold meshes are no longer supported. You can use ``MeshInstance3D`` for
    rendering non-manifold geometry, such as quads or planes.
```
Soften the suggestion a bit since it may not be the correct workaround in all cases

Android
~~~~~~~

.. note::

>     Android sensor events are no longer enabled by default (`GH-94799`_), projects that use sensor events can
>     enable them as needed in Project Settings under "input_devices/sensors/".

Reviewer: ```suggestion
    Android sensor events are no longer enabled by default (`GH-94799`_). Projects that use sensor events can
    enable them as needed in Project Settings under **Input Devices > Sensors**.
```
